networking/eth2/src/test/java/tech/pegasys/teku/networking/eth2/gossip/subnets/StableSubnetSubscriberTest.java

```diff
@@ -17,132 +17,83 @@
 
   @Test
-  void shouldIncreaseNumberOfSubscriptionsWhenNumberOfValidatorsIncrease() {
+  void shouldSubscribeToSubnetsPerNodeAtStart() {
     StableSubnetSubscriber stableSubnetSubscriber = createStableSubnetSubscriber();
-
-    stableSubnetSubscriber.onSlot(ZERO, 0);
-    verifyNoInteractions(validatorApiChannel);
-
-    stableSubnetSubscriber.onSlot(ONE, 3);
-
+    final UInt64 currentSlot = ZERO;
+    stableSubnetSubscriber.onSlot(currentSlot);
     ArgumentCaptor<Set<SubnetSubscription>> subnetSubscriptions =
         ArgumentCaptor.forClass(Set.class);
-    verify(validatorApiChannel).subscribeToPersistentSubnets(subnetSubscriptions.capture());
 
-    assertThat(subnetSubscriptions.getValue()).hasSize(3);
+    verify(attestationTopicSubscriber).subscribeToPersistentSubnets(subnetSubscriptions.capture());
+    assertThat(subnetSubscriptions.getValue())
+        .hasSize(spec.getNetworkingConfig().getSubnetsPerNode());
+    final UInt64 unsubscriptionSlot =
+        spec.getGenesisSpec()
+            .miscHelpers()
+            .calculateNodeSubnetUnsubscriptionSlot(nodeId.get(), currentSlot);
+    subnetSubscriptions
+        .getValue()
+        .forEach(
+            subnetSubscription ->
+                assertThat(subnetSubscription.getUnsubscriptionSlot())
+                    .isEqualTo(unsubscriptionSlot));
     assertSubnetsAreDistinct(subnetSubscriptions.getValue());
   }
 
-  @Test
-  void shouldSubscribeToAllSubnetsWhenNecessary() {
-    StableSubnetSubscriber stableSubnetSubscriber = createStableSubnetSubscriber();
-
-    UInt64 slot = UInt64.valueOf(15);
-    stableSubnetSubscriber.onSlot(slot, attestationSubnetCount + 2);
-
-    ArgumentCaptor<Set<SubnetSubscription>> subnetSubscriptions =
-        ArgumentCaptor.forClass(Set.class);
-    verify(validatorApiChannel).subscribeToPersistentSubnets(subnetSubscriptions.capture());
-    assertThat(subnetSubscriptions.getValue()).hasSize(attestationSubnetCount);
-    assertSubnetsAreDistinct(subnetSubscriptions.getValue());
-    assertUnsubscribeSlotsAreInBound(subnetSubscriptions.getValue(), UInt64.valueOf(15));
-  }
-
-  @Test
-  void shouldStaySubscribedToAllSubnetsEvenIfValidatorNumberIsDecreased() {
-    StableSubnetSubscriber stableSubnetSubscriber = createStableSubnetSubscriber();
-
-    stableSubnetSubscriber.onSlot(ZERO, attestationSubnetCount + 8);
-
-    ArgumentCaptor<Set<SubnetSubscription>> subnetSubscriptions =
-        ArgumentCaptor.forClass(Set.class);
-    verify(validatorApiChannel).subscribeToPersistentSubnets(subnetSubscriptions.capture());
-    assertSubnetsAreDistinct(subnetSubscriptions.getValue());
-    assertThat(subnetSubscriptions.getValue()).hasSize(attestationSubnetCount);
-
-    stableSubnetSubscriber.onSlot(UInt64.valueOf(2), attestationSubnetCount);
-
-    verifyNoMoreInteractions(validatorApiChannel);
-  }
-
   @Test
   void shouldReplaceExpiredSubscriptionsWithNewOnes() {
```

Comment:
I think the comment in the end is no more relevant, since it's calculated now. I'd add following checks to these test: 1. SubnetsSet1 != SubnetsSet2 2. unsubscriptionSlot2  - unsubscriptionSlot1 = 256 epochs

Reply:
Done